### PR TITLE
REF: rework device/pv config storage

### PR DESCRIPTION
**Pull request by klauer**
* One possibility for a path forward; fleshed out to the point of working tests (lack of discussion time? why not code!)
* Not entirely sold on this just yet
* This represents a "verbose config" storage, avoiding the ugly-but-terse "attribute to checklist" style that was previously here
* Since we've decided on providing a configuration GUI for v0, I myself am at least less worried about making the format something a human would _want_ to edit (though this form absolutely is something they _could_ edit).
* Some is still not implemented (methods to search the configuration for attributes or PVs or tags)

(for discussion tomorrow, hopefully, @ZLLentz !)

### CODE REVIEW

File: atef/check.py [hunk 1]
```diff
@@ -494,31 +544,35 @@ def _compare(self, value: Number) -> bool:
...
+@dataclass
+class IdentifierAndComparison:
+    #: PV name, attribute name, or test-specific identifier.
+    identifiers: List[str] = field(default_factory=list)
+    #: The comparisons to perform for *each* of the identifiers.
+    comparisons: List[Comparison] = field(default_factory=list)
 
 
 @dataclass
 class Configuration:
```
Comment (klauer) [inline atef/check.py:556]: `@serialization.as_tagged_union`

File: atef/check.py [hunk 2]
```diff
@@ -527,48 +581,23 @@ class ConfigurationFile:
     A configuration file comprised of a number of devices/PV configurations.
     """
 
-    #: Dictionary of happi device name to DeviceConfiguration.
-    devices: Dict[str, DeviceConfiguration]
+    configs: List[Union[PVConfiguration, DeviceConfiguration]]
```
Comment (klauer) [inline atef/check.py:584]:
```suggestion
    configs: List[Configuration]
```

File: atef/check.py [hunk 3]
```diff
@@ -494,31 +544,35 @@ def _compare(self, value: Number) -> bool:
         return True
 
 
-ItemToChecks = Dict[
-    str,
-    Union[Comparison, Sequence[Comparison]],
-]
+@dataclass
+class IdentifierAndComparison:
```
Comment (ZLLentz) [inline atef/check.py:548]: Do these need a `name` field so we can identify them against the other checklist items in `Configuration`

Comment (klauer) [reply atef/check.py:552]: Added in [4468a7c](https://github.com/pcdshub/atef/pull/27/commits/4468a7c979b5cceff488845ca45af05003d25317)

**Review by ZLLentz — APPROVED** [review]
I tried to look through this closely, the parts that I understood all the context for looked good. I think we should roll with this and consider revisiting depending on if issues crop up while making the editor.

**Comment by klauer** [comment]
Sounds good! Clicking the button.

**Pull request merged**